docs/se/health.adoc

```diff
@@ -76,46 +77,161 @@ Optional dependency to use built-in health checks:
+You can create custom health checks which reflect those conditions and add them to Helidon's overall health assessment of your application.
+
+A health check is a Java functional interface that returns a new
+`HealthCheckResponse` instance each time Helidon queries the health check.
+Each health check also has a fixed name and a fixed health check type (start-up, liveness, or readiness).
 
-A health check is a Java functional interface that returns a
-`HealthCheckResponse` instance. You can create custom health checks in several ways, among them:
+Your code registers a custom health check by invoking a method on Helidon-provided types in one of the following ways:
 
-* write a class which implements the `HealthCheck` interface,
-* write an in-line lambda expression, or
-* use a method reference a method with the double colon operator `::`.
+* Pass the name and type of the health check and a `Supplier` of a `HealthCheckResponse` such as a method reference or a lambda expression.
+* Pass an instance of a class which implements the `HealthCheck` interface.
+
+Within an application different techniques might make sense for different custom health checks, depending on the complexity of the logic for computing the status for each check.
+The various styles are functionally equivalent; for a given custom health check choose the style which enhances the readability and clarity of your code.
+The examples below, in no particular order, implement the same custom health check functionality in different ways to illustrate.
+
+==== Option 1: Using a `HealthCheckResponse` supplier method
+If you gather the logic for computing the health check response into a method you can use a method reference to register the health check.
 
 [source,java]
-.Health check with a lambda expression:
+.Declaring a health check response supplier method
 ----
-Supplier<HealthCheckResponse> hcr = () -> HealthCheckResponse
-                .builder()
-                .detail("exampleValue", "looking-good")
-                .status(HealthCheckResponse.Status.UP) // always "up"
-                .build();
+static HealthCheckResponse slowStartLivenessResponse() {
+    long now = System.currentTimeMillis();
+    return HealthCheckResponse.builder()
+            .detail("time", now)
+            .status(now - serverStartTime >= 8000)
+            .build();
+}
+----
+
+[source,java]
+.Registering a health check using a method reference
+----
+ObserveFeature observe = ObserveFeature.builder()
+                             .config(config.get("server.features.observe")) // <1>
+                             .addObserver(HealthObserver.builder()          // <2>
+                             .useSystemServices(true) // Include Helidon-provided health checks.
+                             .addCheck(Main::slowStartLivenessResponse, // <3>
+                                       HealthCheckType.READINESS,       // <4>
+                                       "live-after-8-seconds")          // <5>
```

Comment:
Method name has "Liveness" in it but type is READINESS?

Reply:
Good catch. I also reformatted this snippet to rein in the indentation a bit!